clang build broken, vector<bool> issue

unpack:
```
cyberway/libraries/fc/include/fc/io/raw.hpp:248:10: error: no matching constructor for initialization of 'std::__1::__bit_reference<std::__1::vector<bool, std::__1::allocator<bool> >, true>'
       T tmp;
         ^
cyberway/libraries/fc/include/fc/io/raw.hpp:577:18: note: in instantiation of function template specialization 'fc::raw::unpack<fc::datastream<const char *>, std::__1::__bit_reference<std::__1::vector<bool,
      std::__1::allocator<bool> >, true> >' requested here
        fc::raw::unpack( s, *itr );
                 ^
cyberway/libraries/chain/abi_serializer.cpp:28:16: note: in instantiation of function template specialization 'fc::raw::unpack<fc::datastream<const char *>, bool>' requested here
      fc::raw::unpack( stream, temp );
               ^
cyberway/libraries/chain/abi_serializer.cpp:37:23: note: in instantiation of function template specialization 'eosio::chain::variant_from_stream<std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
               return variant_from_stream<vector<T>>(stream);
                      ^
/usr/local/Cellar/llvm@4/4.0.1/bin/../include/c++/v1/__bit_reference:35:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 0 were provided
class __bit_reference
      ^
/usr/local/Cellar/llvm@4/4.0.1/bin/../include/c++/v1/__bit_reference:72:5: note: candidate constructor not viable: requires 2 arguments, but 0 were provided
    __bit_reference(__storage_pointer __s, __storage_type __m) _NOEXCEPT
    ^
```

to_variant:
```
cyberway/libraries/fc/include/fc/reflect/variant.hpp:67:28: error: no member named 'visit' in 'fc::reflector<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >'
         fc::reflector<T>::visit( to_variant_visitor<T>( mvo, v ) );
         ~~~~~~~~~~~~~~~~~~^
cyberway/libraries/fc/include/fc/reflect/variant.hpp:100:52: note: in instantiation of function template specialization
      'fc::if_enum<fc::false_type>::to_variant<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >' requested here
      if_enum<typename fc::reflector<T>::is_enum>::to_variant( o, v );
                                                   ^
cyberway/libraries/fc/include/fc/variant.hpp:561:7: note: in instantiation of function template specialization
      'fc::to_variant<std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > > >' requested here
      to_variant( val, *this );
      ^
cyberway/libraries/fc/include/fc/variant.hpp:532:21: note: in instantiation of function template specialization 'fc::variant::variant<std::__1::__bit_const_reference<std::__1::vector<bool,
      std::__1::allocator<bool> > > >' requested here
          vars[i] = variant(t[i]);
                    ^
cyberway/libraries/fc/include/fc/variant.hpp:561:7: note: in instantiation of function template specialization 'fc::to_variant<bool>' requested here
      to_variant( val, *this );
      ^
cyberway/libraries/chain/abi_serializer.cpp:29:14: note: in instantiation of function template specialization 'fc::variant::variant<std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
      return fc::variant(temp);
             ^
cyberway/libraries/chain/abi_serializer.cpp:37:23: note: in instantiation of function template specialization 'eosio::chain::variant_from_stream<std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
               return variant_from_stream<vector<T>>(stream);
                      ^
```

--- a/libraries/chain/include/eosio/chain/fc_pack.hpp
+++ b/libraries/chain/include/eosio/chain/fc_pack.hpp
@@ -6,10 +6,12 @@ namespace fc {
     class variant;
 
     template<typename T>
-    inline void to_variant(const chainbase::oid<T>& var,  variant& vo);
+    inline void to_variant(const chainbase::oid<T>& var, variant& vo);
 
     template<typename T>
-    inline void from_variant(const variant& var,  chainbase::oid<T>& vo);
+    inline void from_variant(const variant& var, chainbase::oid<T>& vo);
+
+    inline void to_variant(const std::vector<bool>& t, variant& vo);
 
     namespace raw {
         template<typename Stream, typename T>
@@ -19,7 +21,7 @@ namespace fc {
         inline void unpack(Stream& s, chainbase::oid<T>& o);
 
         template<typename Stream>
-        inline void unpack(Stream& s, typename std::vector<bool>::reference& o);
+        inline void unpack(Stream& s, std::vector<bool>& value);
     } // namespace raw
 } // namespace fc
 
@@ -27,13 +29,22 @@ namespace fc {
 
 namespace fc {
     template<typename T>
-    inline void to_variant(const chainbase::oid<T>& var,  variant& vo) {
+    inline void to_variant(const chainbase::oid<T>& var, variant& vo) {
         vo = var._id;
     }
 
     template<typename T>
-    inline void from_variant(const variant& var,  chainbase::oid<T>& vo) {
+    inline void from_variant(const variant& var, chainbase::oid<T>& vo) {
         vo._id = var.as_int64();
+    }
+
+    inline void to_variant(const std::vector<bool>& vect, variant& v) {
+        std::vector<variant> vars(vect.size());
+        size_t i = 0;
+        for (bool b : vect) {
+            vars[i++] = variant(b);
+        }
+        v = std::move(vars);
     }
 
     namespace raw {
@@ -49,10 +60,20 @@ namespace fc {
 
         /** std::vector<bool> has custom implementation and pack bools as bits */
         template<typename Stream>
-        inline void unpack(Stream& s, typename std::vector<bool>::reference& o) {
-            bool v;
-            fc::raw::unpack(s, v);
-            o = v;
+        inline void unpack(Stream& s, std::vector<bool>& value) {
+            // TODO: can serialize as bitmap to save up to 8x storage (implement proper pack)
+            unsigned_int size;
+            fc::raw::unpack(s, size);
+            FC_ASSERT(size.value <= MAX_NUM_ARRAY_ELEMENTS);
+            value.resize(size.value);
+            auto itr = value.begin();
+            auto end = value.end();
+            while (itr != end) {
+                bool b = true;
+                fc::raw::unpack(s, b);
+                *itr = b;
+                ++itr;
+            }
         }
     } // namespace raw
 } // namespace fc

